rotkehlchen/exchanges/coinbase.py

```diff
@@ -684,6 +686,98 @@ def _process_normal_trade(self, event: dict[str, Any], trades: list[Trade]) -> N
 
         return None
 
+    def _process_advanced_trade(self, event: dict[str, Any], trades: list[Trade]) -> None:
+        """Turns an advanced_trade_fill transaction into a rotki trade and adds it to the list.
+
+        https://docs.cloud.coinbase.com/sign-in-with-coinbase/docs/api-transactions#transaction-types
+        If the coinbase transaction is not a trade related transaction nothing happens.
+        """
+        try:
+            if event['status'] != 'completed':
+                # We only want to deal with completed trades
+                return None
+
+            # Contrary to the Coinbase documentation we will use created_at, and never
+            # payout_at. It seems like payout_at is not actually the time when the
+            # trade is settled. Reports generated by Coinbase use created_at as well
+            if event.get('created_at') is not None:
+                raw_time = event['created_at']
+            else:
+                raw_time = event['payout_at']
+
+            timestamp = deserialize_timestamp_from_date(raw_time, 'iso8601', 'coinbase')
+            trade_type = TradeType.deserialize(event['advanced_trade_fill']['order_side'])
+
+            if trade_type not in (TradeType.BUY, TradeType.SELL):
+                return None
+
+            # Notice that we do not use abs() yet
+            amount = deserialize_asset_amount(event['amount']['amount'])
+
+            # Each trade has two sides, we ignore one of them
+            if ((trade_type == TradeType.SELL and amount > 0) or
+                    (trade_type == TradeType.BUY and amount < 0)):
+                return None
+
+            tx_asset_identifier = event['amount']['currency']
+            tx_asset = asset_from_coinbase(tx_asset_identifier, time=timestamp)
+            asset_identifiers = event['advanced_trade_fill']['product_id'].split('-')
+            other_side_identifier = (asset_identifiers[0]
+                                     if asset_identifiers[0] != tx_asset_identifier
+                                     else asset_identifiers[1])
```

Comment:
Can you catch a possible indexError? If the product for some reason doesn't follow the `ASSET1-ASSET2` format the split will return a list with a single element and won't be correctly indexed. Also not sure if there are symbols with a `-`, something like `BTC-L`. Can you add a check to ensure that you have the expected number of symbols and if not log an error and exit?